grades/management/commands/adjust_exam_grades_from_csv.py

```diff
@@ -0,0 +1,98 @@
+"""
+Freezes final grades for a course
+"""
+import csv
+import argparse
+from collections import namedtuple
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.management import BaseCommand, CommandError
+
+from grades.models import ProctoredExamGrade
+
+
+class ParsingError(CommandError):
+    """Custom class for parsing exceptions"""
+    pass
+
+
+class GradeRowParser:
+    """Parser for rows of grade adjustment information in a CSV"""
+    RowProps = namedtuple('RowProps', ['exam_grade_id', 'score'])
+    default_col_names = dict(
+        exam_grade_id='proctoredexam_id',
+        score='score',
+    )
+
+    def __init__(self, col_names=None):
+        col_names = col_names or {}
+        self.col_names = self.RowProps(**{**self.default_col_names, **col_names})
+
+    def parse_and_validate_row(self, row):
+        """Parses a row of grade adjustment info and makes sure it doesn't contain bad data"""
+        try:
+            parsed_row = self.RowProps(
+                exam_grade_id=int(row[self.col_names.exam_grade_id]),
+                score=float(row[self.col_names.score]),
+            )
+        except KeyError as e:
+            raise ParsingError('Row is missing a required column: {}'.format(str(e)))
+
+        if parsed_row.score < 0.0 or parsed_row.score > 100.0:
+            row_identifier = '{}: {}'.format(self.col_names.exam_grade_id, parsed_row.exam_grade_id)
+            raise ParsingError('[{}] "score" value must be between 0 and 100'.format(row_identifier))
+        return parsed_row
+
+
+class Command(BaseCommand):
+    """Parses a csv with exam grade adjustment information and changes the appropriate grades"""
+    help = "Parses a csv with exam grade adjustment information and changes the appropriate grades"
+
+    def add_arguments(self, parser):
+        parser.add_argument('csvfile', type=argparse.FileType('r'), help='')
+        parser.add_argument(
+            '--grade-id-col-name',
+            default=GradeRowParser.default_col_names['exam_grade_id'],
+            help='Name of the column that contains the proctored exam grade id')
+        parser.add_argument(
+            '--score-col-name',
+            default=GradeRowParser.default_col_names['score'],
+            help='Name of the column that contains the score value'
+        )
+
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument,too-many-locals
+        col_names = dict(
+            exam_grade_id=kwargs.get('grade_id_col_name'),
+            score=kwargs.get('score_col_name'),
+        )
+        csvfile = kwargs.get('csvfile')
+        reader = csv.DictReader(csvfile.read().splitlines(), delimiter='\t')
+        grade_row_parser = GradeRowParser(col_names=col_names)
+
+        total_rows = 0
+        grades_changed = 0
+        grades_unchanged = 0
+        not_found = 0
+        for row in reader:
+            parsed_row = grade_row_parser.parse_and_validate_row(row)
```

Comment:
I would suggest we parse and validate all the rows upfront because if one part of the CSV is invalid it casts doubt on the rest of it.